spec/index_spec.rb

```diff
@@ -372,4 +372,51 @@
       )
     end
   end
+
+  describe Daru::CategoricalIndex do
+    before do
+      @idx = Daru::CategoricalIndex.new(
```

Comment:
@lokeshh With pleasure.
Look.

1) The first usage of `subject` is for semantics: modern rspec implies small contexts, each of them testing exactly one test subject (if possible, of course, which is not always the case). Like this:

```ruby
describe Daru::CategoricalIndex do
  let(:idx) { Daru::CategoricalIndex.new(...) } # <- that's some data we use...
  context '#[]' do
    subject { idx[:a] } # that's the value we will be checking in this context
    # or, subject could be named:
    subject(:one_category) { idx[:a] } # that's the value we will be checking in this context
  end
end
```

2) In many cases, when you have subject, you can write simpler `it`s, like this (assuming we are inside context described above):
```ruby
it { is_expected.to be_a Daru::Index }
it { is_expected.to eq Daru::Index.new([0, 2, 3]) }
```
^ rspec even wil format that properly, without any text description (if such test fails, it will write something like "In test 'expected to be an instance of Daru::Index class'")

3) With `rspec-its` gem (which I also strongly suggest -- in any case, I'm adding it in my "refactor" branch), the "implicit subject" tests can be even more powerful, like this:

```ruby
its(:size) { is_expected.to eq 3 }
its(:first) { is_expected.to eq 0 }
....
```

Such "atomic" checks in many cases are easier to write and rerun, and find the problems (in current Daru specs, it is sometimes complicated fails like "(this huge dataframe) is expected to be equal to (this huge dataframe), go figure the difference!"

So, summing it up: a) semantics of specs (focus on "we test this specific value") + b) more concise and DRY specs